reconcile/utils/models.py

```diff
@@ -0,0 +1,125 @@
+from collections.abc import (
+    Callable,
+    Generator,
+    MutableMapping,
+)
+from typing import Any
+
+from croniter import croniter
+from pydantic import (
+    BaseModel,
+    ValidationError,
+)
+from pydantic import errors as pydantic_errors
+from pydantic.fields import ModelField
+
+
+def data_default_none(
+    klass: type[BaseModel], data: MutableMapping[str, Any]
+) -> MutableMapping[str, Any]:
+    """Set default values to None for required but optional fields."""
```

Comment:
Can you please explain this a little more? How is something required, and optional at the same time?

Reply:
No, I just found the comment confusing, perhaps rephrase it: `Set default values to None for fields required in klass, where keys are missing in data.`